ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform.c

```diff
@@ -101,44 +109,43 @@ VL53L0X_Error VL53L0X_UnlockSequenceAccess(VL53L0X_DEV Dev){
     return Status;
 }
 
-// the ranging_sensor_comms.dll will take care of the page selection
 VL53L0X_Error VL53L0X_WriteMulti(VL53L0X_DEV Dev, uint8_t index, uint8_t *pdata, uint32_t count){
 
     VL53L0X_Error Status = VL53L0X_ERROR_NONE;
     int32_t status_int = 0;
-	uint8_t deviceAddress;
+    uint8_t I2CBuffer[count+1];
 
     if (count>=VL53L0X_MAX_I2C_XFER_SIZE){
         Status = VL53L0X_ERROR_INVALID_PARAMS;
     }
 
-	deviceAddress = Dev->I2cDevAddr;
+    I2CBuffer[0] = index;
+    memcpy(&I2CBuffer[1], pdata, count);
 
-	status_int = VL53L0X_write_multi(deviceAddress, index, pdata, count);
+    status_int = i2c_write(Dev->i2c, I2CBuffer, count+1, Dev->I2cDevAddr);
 
-	if (status_int != 0)
-		Status = VL53L0X_ERROR_CONTROL_INTERFACE;
+    if (status_int < 0){
+        Status = VL53L0X_ERROR_CONTROL_INTERFACE;
+         SYS_LOG_ERR("Failed to write");
+    }
 
     return Status;
 }
 
-// the ranging_sensor_comms.dll will take care of the page selection
 VL53L0X_Error VL53L0X_ReadMulti(VL53L0X_DEV Dev, uint8_t index, uint8_t *pdata, uint32_t count){
     VL53L0X_I2C_USER_VAR
     VL53L0X_Error Status = VL53L0X_ERROR_NONE;
     int32_t status_int;
-	uint8_t deviceAddress;
 
     if (count>=VL53L0X_MAX_I2C_XFER_SIZE){
         Status = VL53L0X_ERROR_INVALID_PARAMS;
     }
 
-    deviceAddress = Dev->I2cDevAddr;
-
-	status_int = VL53L0X_read_multi(deviceAddress, index, pdata, count);
-
-	if (status_int != 0)
-		Status = VL53L0X_ERROR_CONTROL_INTERFACE;
+    status_int = i2c_burst_read(Dev->i2c, Dev->I2cDevAddr, index, pdata, count);
+    if (status_int < 0) {
+        SYS_LOG_ERR("Failed to read");
+        return -EIO;
+    }
 

```

Comment:
why modify all this, why not just provide a version of VL53L0X_read_multi?

Reply:
same as previous remark